### ISSUE TITLE
feat: brier score for uncertainty estimation on classification

### ISSUE THREAD

**Issue by xuyxu** [PyTorchLightning/metrics#946]
### Motivation

Brier score ([WIKI](https://en.wikipedia.org/wiki/Brier_score)) is a popular metric used for uncertainty estimation, an important machine learning task that receives growing interests recently. See this paper for details: https://proceedings.neurips.cc/paper/2017/file/9ef2ed4b7fd2c810847ffa5fa85bce38-Paper.pdf.

It is also a metric already implemented in [scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.brier_score_loss.html), although the [multi-class version](https://github.com/scikit-learn/scikit-learn/pull/22046) is still under development. A brier score metirc that could be directly used in Pytorch would be quite useful, and I am willing to work on it if this feature request is approved. 

### Pitch

A brier score metric in the [classification](https://torchmetrics.readthedocs.io/en/latest/references/modules.html#classification) part.

**Comment by SkafteNicki** [comment]
Hi @xuyxu,
Sounds like a great addition to Torchmetrics, feel free to send a PR with full (or partly) implementation of the metric and then we can help you get it merged :]

**Comment by cemde** [comment]
isnt this just an alias for the MeanSquaredError?

**Comment by xuyxu** [comment]
> isnt this just an alias for the MeanSquaredError?

Hi @cemde, I agree with you from the point of formula. Essentially,  the brier score is the MSE between the one-hot encoded classification label and the probability output. However, after looking at the function sigature of [MSE](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/functional/regression/mse.py#L56) and classification-related metrics, e.g., [ACC](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/functional/classification/accuracy.py#L258), I think some extra steps are necessary to make it a classification metric that performs well across all cases.

**Comment by xuyxu** [comment]
After a thorough look at the internal methods, I think it is quite easy to compute brier score based on existing metrics, therefore no need to add a new metric and inevitably bring some maintenance costs. Thanks!
